Update docs for 7.0.1 otel metrics (#432)
DuendeSoftware/docs.duendesoftware.com: IdentityServer/v7/docs/content/diagnostics/otel/metrics.md

(added in v7.0)

> The metric counters are designed to provide a high level overview. They are designed to not contain any
> sensitive information. The counters may contain tags to indicate the source of the data. Metrics are
> intended to be used to provide overall health metrics and are typically used to show graphs on

Reviewer: I think we should move the remarks about sensitive data and tags later (see below), and start off with a statement of what a metric is.
```suggestion
OpenTelemetry metrics are counters designed to provide a high level overview of the frequency events that are occurring in your system. 
Metrics are intended to be used to provide overall health metrics and are typically used to show graphs on
```

a dashboard or to set up monitoring rules. If the metrics indicates that something has happened,

> the traces and logs are used to find out the reason. Open Telemetry monitoring tools might have

Reviewer: ```suggestion
the traces and logs are used to find out the reason. Open Telemetry monitoring tools might
```

provide features to find the traces and logs corresponding to certain metrics.

IdentityServer emits metrics from the IdentityServer library. Our quick start for the UI also

> [contains metrics](#metrics-in-the-ui) that can be used as a starting point for monitoring UI events.

Reviewer: I would move the "no sensitive information" and "may contain flags" to the end of this paragraph:

The metric counters that IdentityServer emits are designed to not contain any sensitive information. They often are tagged to indicate the source of the events.

## High level Metrics
These metrics are instrumented by the IdentityServer middleware and services and are
intended to describe the overall usage and health of the system. They could provide the
starting point for building a metrics dashboard. The high level metrics are enabled through
the `Duende.IdentityServer.Telemetry.ServiceName` meter.

#### Telemetry.Metrics.Counters.Operation
Counter name: *tokenservice.operation*

Aggregated counter of failed and successful operations. Note that it is expected to have some failures during

> normal operation of IdentityServer. There should be no internal_errors though, they indicate that there are

Reviewer: What are internal_errors? Is that a tag?

Oh, I guess it is a possible value for the result tag. Maybe we could clarify that in the text.

Also, just a nit, but this is a comma splice.

> unhandled exceptions. The failure/success ratio can be used as a very high level health metric.

Reviewer: ```suggestion
unhandled exceptions. The error/success ratio can be used as a very high level health metric.
```

(Since we don't have a failure event anymore, and the result will either be success or error)

|Tag|Description|
|---|---|
|error | Error label on errors|
|result | Success, error or internal_error |
|client | Id of client requesting the operation. May be empty. |

#### Telemetry.Metrics.Counters.ActiveRequests
Counter name: *active_requests*

Gauge/up-down counter that shows current active requests that are processed by any IdentityServer endpoint. 
Note that the pages in the user interface are not IdentityServer endpoints and are not included in this count.
This metrics
|Tag|Description|
|---|---|
|endpoint | The type name for the endpoint processor |
|path | The path of the request |

> ### Detailed Metrics - Experimental

Reviewer: Our style sheet makes h3 and h4 indistinguishable. I try to avoid h3s. (We probably should change the style sheet, come to think of it, but that's a bigger change than we're doing here obviously).

These detailed metrics are instrumented by the IdentityServer middleware and services and track usage of specific

> flows and features. These metrics are enabled through the `Duende.IdentityServer.Telemetry.ServiceName.Experimental`

Reviewer: I don't think ServiceName is actually part of the meter name? Maybe I'm misunderstanding the naming conventions...

> flows and features. These metrics are enabled through the `Duende.IdentityServer.Telemetry.ServiceName.Experimental`

Reviewer: I'm also not really sure how to talk about the relationship between metrics and meters, but "enabled through" doesn't sound quite right to me.
